desktop/ui/src/main/java/org/datacleaner/widgets/options/RemoteComponentsOptionsPanel.java

```diff
@@ -0,0 +1,197 @@
+        addField("Password", passwordTextField, applyButton);
+    }
+
+    private void addField(String labelText, JComponent... fields) {
+        row++;
+
+        JPanel spaceFiller = new JPanel();
+        spaceFiller.setBackground(WidgetUtils.COLOR_DEFAULT_BACKGROUND);
+        WidgetUtils.addToGridBag(new JLabel(labelText), this, 0, row, 1, 1, GridBagConstraints.LINE_START, padding,
+                weightx, weighty, GridBagConstraints.HORIZONTAL);
+        for (int i = 0; i < fields.length; i++) {
+            WidgetUtils.addToGridBag(fields[i], this, 1 + i, row, 1, 1, GridBagConstraints.LINE_START, padding, weightx,
+                    weighty, GridBagConstraints.HORIZONTAL);
+        }
+        WidgetUtils.addToGridBag(spaceFiller, this, 1 + fields.length, row, 1, 1, GridBagConstraints.LINE_END, padding,
+                maxWeightx, weighty, GridBagConstraints.REMAINDER);
+    }
+
+    private Component getDescriptionComponent() {
+        DCHtmlBox htmlBox = new DCHtmlBox("This dialog is for credentials setting of users registered at "
+                + "<a href=\"http://datacleaner.org\">datacleaner.org</a>. <br><br>"
+                + "Remote components are a cloud service providing new functions. "
+                + "These remote components run at the server, consume provided input data and return the results. ");
+
+        return htmlBox;
+    }
+
+    private Resource getDataCleanerConfigurationFileResource() {
+        final RepositoryFile configurationFile = _configuration.getHomeFolder().toRepositoryFolder()
+                .getFile(DataCleanerConfigurationImpl.DEFAULT_FILENAME);
+        final Resource resource = configurationFile.toResource();
+        return resource;
+    }
+
+    private void updateDcConfiguration() {
+        final RemoteServerData existingServerData = getServerData();
+        final String username = usernameTextField.getText();
+        final String password = new String(passwordTextField.getPassword());
+
+        final RemoteServerDataImpl newServerData = new RemoteServerDataImpl(existingServerData.getUrl(),
+                existingServerData.getServerName(), existingServerData.getServerPriority(), username, password);
+        final RemoteServerConfiguration remoteServerConfiguration = _configuration.getEnvironment()
+                .getRemoteServerConfiguration();
+        if (remoteServerConfiguration.getServerList().isEmpty()) {
+            remoteServerConfiguration.getServerList().add(newServerData);
+        } else {
+            remoteServerConfiguration.getServerList().remove(0);
+            remoteServerConfiguration.getServerList().add(0, newServerData);
+        }
+
+        final DataCleanerConfigurationUpdater configurationUpdater = new DataCleanerConfigurationUpdater(
+                getDataCleanerConfigurationFileResource());
+        configurationUpdater.update("descriptor-providers:remote-components:server:username", username);
```

Comment:
Yea that would be quite a lot prettier ... In general I don't like that "Updater" class because it is just a glorified DOM editor. It has not knowledge about the XSD of conf.xml. So for now I consider it just throw-away-ware until something better comes along.